csrc/device_lower/pass/unroll.cpp

```diff
@@ -65,8 +65,10 @@ void UnrollPass::dispatch(Expr* expr) {
 
   // Predicate MBarrierWaitParity is required for 1D TMA.
   if (one_dim_tma_predicate_added_ && expr->isA<kir::MBarrierWaitParity>() &&
-      for_loops_.back()->circularBufferLoopStage() ==
-          CircularBufferLoopStage::ComputeWarp) {
+      std::any_of(for_loops_.begin(), for_loops_.end(), [](const ForLoop* fl) {
```

Comment:
I think this would be a useful utility:
```cpp
namespace ir_utils {

bool isCircularBufferType(const std::vector<ForLoop*> for_loops, CircularBufferLoopStage stage_type) {
  return std::any_of(for_loops.begin(), for_loops.end(), [stage_type](const ForLoop* fl) {
        return fl->circularBufferLoopStage() == stage_type;
      };
}
```

} // namespace ir_utils